plugin/src/commonObjects.ts

```diff
@@ -1,63 +1,141 @@
 export const globals = new Set([
-  'this',
-  'console',
-  'performance',
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects
+
+  // Value properties
+  'globalThis',
+  'Infinity',
+  'NaN',
+  'undefined',
+
+  // Function properties
+  'eval',
+  'isFinite',
+  'isNaN',
+  'parseFloat',
+  'parseInt',
+  'decodeURI',
+  'decodeURIComponent',
+  'encodeURI',
+  'encodeURIComponent',
+  'escape',
+  'unescape',
+
+  // Fundamental objects
+  'Object',
+  'Function',
+  'Boolean',
+  'Symbol',
+
+  // Error objects
+  'Error',
+  'AggregateError',
+  'EvalError',
+  'RangeError',
+  'ReferenceError',
+  'SyntaxError',
+  'TypeError',
+  'URIError',
+  'InternalError',
+
+  // Numbers and dates
+  'Number',
+  'BigInt',
+  'Math',
   'Date',
+
+  // Text processing
+  'String',
+  'RegExp',
+
+  // Indexed collections
   'Array',
-  'ArrayBuffer',
   'Int8Array',
-  'Int16Array',
-  'Int32Array',
   'Uint8Array',
   'Uint8ClampedArray',
+  'Int16Array',
   'Uint16Array',
+  'Int32Array',
   'Uint32Array',
+  'BigInt64Array',
+  'BigUint64Array',
   'Float32Array',
   'Float64Array',
-  'HermesInternal',
+
+  // Keyed collections
+  'Map',
+  'Set',
+  'WeakMap',
+  'WeakSet',
+
+  // Structured data
+  'ArrayBuffer',
+  'SharedArrayBuffer',
+  'DataView',
+  'Atomics',
   'JSON',
-  'Math',
-  'Number',
-  'Object',
-  'String',
-  'Symbol',
-  'undefined',
+
+  // Managing memory
+  'WeakRef',
+  'FinalizationRegistry',
+
+  // Control abstraction objects
+  'Iterator',
+  'AsyncIterator',
+  'Promise',
+  'GeneratorFunction',
+  'AsyncGeneratorFunction',
+  'Generator',
+  'AsyncGenerator',
+  'AsyncFunction',
+
+  // Reflection
+  'Reflect',
+  'Proxy',
+
+  // Internationalization
+  'Intl',
+  // 'Intl.Collator',
```

Comment:
Why is this commented out?

Reply:
If `Intl.Collator` appears in worklet body, our plugin will capture only `Intl`, thus it doesn't make sense to allowlist `Intl.Collator`.